### ISSUE TITLE
Color contrast miscalculates foreground color with opacity

### ISSUE THREAD

**Issue by straker** [dequelabs/axe-core#3924]
Given the following HTML:

```html
<style>
button {
  color: #fff;
  background: #00633D;
  border-color: #00633D;
  opacity: .65;
}
</style>
<button><span>Hello</span></button>
```

The foreground color is calculated as `#c5dcd3`, which is not quite correct. If we put the calculated color next to the rendered HTML, the calculated foreground color is not the same.

<img alt="" src="https://user-images.githubusercontent.com/2433219/221213323-48d7a1cd-dcc4-41f7-bc90-362554763931.png"/>

I'll need to dig into the color spec again to see how opacity works on text nodes as it appears it should use the background behind the element with opacity instead of blending into all backgrounds.

**Comment by padmavemulapati** [comment]
Validated with the latest develp brranch code base,
for the code snippet:
```
<style>
    button {
      color: #fff;
      background: #00633D;
      border-color: #00633D;
      opacity: .65;
    }
    </style>
     <button><span>Hello</span></button>
     <br>
     <br>
     <div id="target" style="height:20px; width:15px; color:rgb(0 0 128); background-color:green;">Hello World</div>
   ```
1. foregrond color (text color)is calculated as- `#FFFFFF` and the background color as `#6A9A81` then it is passing at only for `wcagaa` - `Large Text,UI Components, & Graphical Objects (3:1)` and remaining all are failing for SC - "Regular Text(1.4.3a)"

<img width="692" alt="image" src="https://user-images.githubusercontent.com/47409579/230814445-8058b9ac-e8b6-4964-9f0a-14162660531b.png">

2. When this border-color color is - `#6A9A81` and background color also same then "Regular Text(1.4.3a) fialing in all the onditions.
![image](https://user-images.githubusercontent.com/47409579/230815022-ab512b89-1003-4ff6-a248-1b0c8376dc31.png)

3. for the color:rgb(0 0 128); background-color:green; also 1.4.3a failing on all the conditions.
![image (2)](https://user-images.githubusercontent.com/47409579/230815141-b3575e4f-887c-4a48-a8b1-1a64afe472ac.png)
